scripts/get-crossreg-titles.R

```diff
@@ -0,0 +1,144 @@
+  select(nct_id, official_title) |>
+  rename(id = nct_id) |>
+  rename(official = official_title)
+
+# If there is no official title available in ClinicalTrials.gov, the brief title will be used as the 'title'
+iv_updated_titles <- left_join(iv_ids, nct_full_titles, by = "id") |>
+  mutate(title = coalesce(official, brief)) |>
+  select(-brief, -official)
+
+# Process IV titles to make compatible with title matching algorithm
+iv_updated_titles <- iv_updated_titles |>
+  mutate(title_processed = tolower(title) |>
+           stringr::str_squish() |> # remove whitespace at start and end, as well as any "\t" whitespace characters
+           stringr::str_remove_all("[:punct:]") |>
+           stringr::str_remove_all(" ")) |>
+  select(id, title, title_processed)
+
+###########################################################
+
+# Extract EU IDs and country data (for filtering), plus process EU titles to be used in title matching algorithm
+eu_ids <- eu_protocol_dump |>
+          select(eudract_number,
+                 member_state_concerned,
+                 full_title_of_the_trial,
+                 sponsors)
+
+eu_ids <- rename(eu_ids, id = eudract_number, state = member_state_concerned, title = full_title_of_the_trial)
+
+# Filter out EU trials which dont have 'Germany' listed in member_state_concerned
+# Yields table with 13068 unique trials which mention Germany in this field
+# Can filter however we want here.
+
+eu_only_german <- eu_ids |>
+                  filter(grepl('Germany', state))
+
+# Process EU titles
+eu_only_german <- eu_only_german |> select(id,title) |>
+  tidyr::drop_na(title) |>
+  distinct(id, title,.keep_all = TRUE) |>
+  mutate(title_processed = tolower(title) |>
+           stringr::str_squish() |> # remove whitespace at start and end, as well as any "\t" whitespace characters
+           stringr::str_remove_all("[:punct:]") |>
+           stringr::str_remove_all(" "))
+
+##########################################################
+## Title matching (from 02_cross-reg-title-matching.R)
+
+# assigning maxDist value for title matching (Elements in x will not be matched
+# with elements of table if their distance is larger than maxDist)
+DISTANCE <- 10
+
+## The below takes about 3.5 hours on my baseline 2020 MacBook Air
+## Let it run, it takes a while!!
+
+start_time <- Sys.time()
+
+# use amatch function to find title matches
+# Default method used for matching. Documentation for amatch function:
+# https://www.rdocumentation.org/packages/stringdist/versions/0.9.12/topics/amatch
+ title_matches <- cbind(
+  euctr_title_tm = eu_only_german$title,
+  euctr_id = eu_only_german$id,
+  iv_updated_titles[amatch(
+    eu_only_german$title_processed,
+    iv_updated_titles$title_processed,
+    maxDist = DISTANCE, matchNA = FALSE),]
+) |>
+  # keep only those where a match was found
+  filter(
+    !is.na(id)
+  ) |>
+  distinct(
+    # euctr_id, # we want to preserve all EUCTR crossreg for given trial
+    id,
```

Comment:
@lucfernandez I think we had a comment about this before, but why not preserve all EUCTR crossreg for a given trial? Only applies to two trials, but conceptually I think we should keep these (so uncomment). Thoughts?